test/unit/alignment/matrix/debug_stream_debug_matrix_test.cpp

```diff
@@ -135,75 +135,75 @@ struct debug_matrix_stream_test : public ::testing::Test
+        "T;u   ;u   ;Du  ;Du  ;Du  ;D   ;Dul ;D   ;Dl  ;l   ;l   ;l   ;l   ;l   ;l   ;Dl  ;Dl  ;\n"
+        "A;u   ;Du  ;D   ;Dul ;D   ;Dul ;D   ;u   ;D   ;D   ;Dl  ;l   ;l   ;l   ;l   ;l   ;l   ;\n"
+        "C;u   ;u   ;u   ;D   ;ul  ;D   ;l   ;Dul ;Du  ;Du  ;D   ;D   ;Dl  ;l   ;l   ;l   ;l   ;\n"
+        "G;u   ;u   ;u   ;u   ;D   ;u   ;D   ;l   ;l   ;Dul ;Du  ;Du  ;D   ;D   ;Dl  ;l   ;l   ;\n"
+        "T;N   ;u   ;u   ;u   ;Du  ;Du  ;u   ;D   ;Dl  ;l   ;l   ;Dul ;Du  ;Du  ;D   ;D   ;Dl  ;\n"
     };
 
     std::string trace_matrix_unicode
     {
         " ║ε  ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║   ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        "ε║↺  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║←  ║\n"
+        "ε║↺  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↖  ║↖← ║←  ║↖← ║←  ║←  ║←  ║←  ║↖← ║↖← ║←  ║←  ║←  ║←  ║←  ║←  ║\n"
+        " ║⇡  ║↖  ║↖⇠ ║⇠  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║↖⇠ ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↖  ║↖  ║←  ║↖← ║←  ║←  ║←  ║←  ║←  ║↖← ║↖← ║←  ║←  ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║↖  ║↖  ║⇠  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║↖⇠ ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↖↑ ║↖↑ ║↖  ║↖← ║↖  ║←  ║←  ║←  ║←  ║←  ║←  ║↖← ║↖← ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║↖⇡ ║↖⇡ ║↖  ║↖⇠ ║↖  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║↖⇠ ║↖⇠ ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↖↑ ║↖↑ ║↖↑ ║↖  ║↖↑←║↖  ║↖← ║←  ║←  ║←  ║←  ║←  ║←  ║↖← ║↖← ║\n"
+        " ║⇡  ║⇡  ║↖⇡ ║↖⇡ ║↖⇡ ║↖  ║↖⇡⇠║↖  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║↖⇠ ║↖⇠ ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↖↑ ║↖  ║↖↑←║↖  ║↖↑←║↖  ║↑  ║↖  ║↖  ║↖← ║←  ║←  ║←  ║←  ║←  ║←  ║\n"
+        " ║⇡  ║↖⇡ ║↖  ║↖⇡⇠║↖  ║↖⇡⇠║↖  ║⇡  ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↑  ║↖  ║↑← ║↖  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║←  ║←  ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║⇡  ║↖  ║⇡⇠ ║↖  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↑  ║↑  ║↖  ║↑  ║↖  ║←  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║⇡  ║⇡  ║↖  ║⇡  ║↖  ║⇠  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↺  ║↑  ║↑  ║↑  ║↖↑ ║↖↑ ║↑  ║↖  ║↖← ║←  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║\n"
+        " ║↺  ║⇡  ║⇡  ║⇡  ║↖⇡ ║↖⇡ ║⇡  ║↖  ║↖⇠ ║⇠  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║\n"
```

Comment:
Why these changes?

Reply:
We had in the first implementation only considered normal gaps and now added versions for gap_open and gap_extension. Since the test only uses gap_extensions it is not visible at this places. But we decided to have the solid arrow for gap_opening and the dotted arrow for gap extension as it seems more natural for the association with gap opening and extension.